assessment.rb

```diff
@@ -2,11 +2,36 @@
 # Assessment answers to:
 
 # question 2:
+person = Licensee.create(surname: 'Bishop', given_name: 'Caren', gender: 'f', height: 63, weight: 132, dob: '1943-09-26', eye_color: 'Brown', hair_color: 'Black')
 
 # question 3:
-
+      Licensee.transaction do
+        CSV.foreach(Rails.root + 'data/licensees.csv',
+                    headers: true) do |licensee_row|
+          licensee = licensee_row.to_hash
+          next if Licensee.exists? licensee
+          Licensee.create! licensee
+        end
+      end
 # question 4:
+licensees = Licensee.where("eye_color = :eye_color and hair_color = :hair_color or hair_color = :second_color", { eye_color: "Hazel", hair_color: "Brown", second_color: "Black" })
```

Comment:
I think this is mostly right, but there's a problem with using logical operations in this way - they resolve from left to right. In other words your query tests
`(eye_color == "Hazel && hair_color: "Brown") || hair_color: "Black"`
rather than
`eye_color == "Hazel && (hair_color: "Brown" || hair_color: "Black")`